packages/storage/amplify_storage_s3/lib/src/amplify_storage_s3_impl.dart

```diff
@@ -13,56 +13,76 @@
 // limitations under the License.
 import 'package:amplify_core/amplify_core.dart';
 import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
+import 'package:meta/meta.dart';
+import 'package:path_provider/path_provider.dart' as path_provider;
 
+/// {@template amplify_storage_s3_.amplify_storage_s3_plugin}
+/// The Flutter Storage S3 plugin for the Amplify Storage Category.
+/// {@endtemplate}
 class AmplifyStorageS3 extends AmplifyStorageS3Dart {
+  /// {@macro amplify_storage_s3_.amplify_storage_s3_plugin}
   AmplifyStorageS3({
     super.delimiter,
     super.prefixResolver,
   });
 
-  // TODO(HuiSF): replace with S3 specific types to the generic parameters
   /// A plugin key which can be used with `Amplify.Storage.getPlugin` to retrieve
   /// a S3-specific Storage category interface.
   static const StoragePluginKey<
-      StorageListOperation,
-      StorageListOptions,
-      StorageGetPropertiesOperation,
-      StorageGetPropertiesOptions,
-      StorageGetUrlOperation,
-      StorageGetUrlOptions,
+      S3StorageListOperation,
+      S3StorageListOptions,
+      S3StorageGetPropertiesOperation,
+      S3StorageGetPropertiesOptions,
+      S3StorageGetUrlOperation,
+      S3StorageGetUrlOptions,
       StorageUploadDataOperation,
       StorageUploadDataOptions,
-      StorageCopyOperation,
-      StorageCopyOptions,
-      StorageMoveOperation,
-      StorageMoveOptions,
-      StorageRemoveOperation,
-      StorageRemoveOptions,
-      StorageRemoveManyOperation,
-      StorageRemoveManyOptions,
-      StorageItem,
+      S3StorageDownloadDataOperation,
+      S3StorageDownloadDataOptions,
+      S3StorageDownloadFileOperation,
+      S3StorageDownloadFileOptions,
+      S3StorageCopyOperation,
+      S3StorageCopyOptions,
+      S3StorageMoveOperation,
+      S3StorageMoveOptions,
+      S3StorageRemoveOperation,
+      S3StorageRemoveOptions,
+      S3StorageRemoveManyOperation,
+      S3StorageRemoveManyOptions,
+      S3StorageItem,
+      S3TransferProgress,
       AmplifyStorageS3Dart> pluginKey = _AmplifyStorageS3PluginKey();
+
+  @protected
+  @override
+  Future<String> getTemporaryDirectoryPath() async {
```

Comment:
Fine for now but this should really be injected. This is technically a public API with `getPlugin`